src/plugins/files/server/file_client/file_metadata_client/adapters/es_index.ts

```diff
@@ -82,11 +81,36 @@ export class EsIndexFilesMetadataClient<M = unknown> implements FileMetadataClie
   }
 
   async get({ id }: GetArg): Promise<FileDescriptor<M>> {
-    const { _source: doc } =
-      (await fetchDoc<FileDocument<M>>(this.esClient, this.index, id, this.indexIsAlias)) ?? {};
+    const { esClient, index, indexIsAlias } = this;
+    let doc: FileDocument<M> | undefined;
+
+    if (indexIsAlias) {
+      doc = (
+        await esClient.search<FileDocument<M>>({
+          index,
+          body: {
+            size: 1,
+            query: {
+              term: {
+                _id: id,
+              },
+            },
+          },
+        })
+      ).hits.hits?.[0]?._source;
+    } else {
+      doc = (
+        await esClient.get<FileDocument<M>>({
+          index,
+          id,
+        })
+      )._source;
+    }
 
     if (!doc) {
-      this.logger.error(`File with id "${id}" not found`);
+      this.logger.error(
```

Comment:
@jloleysens - I think this [addresses your comment](https://github.com/elastic/kibana/pull/153342#discussion_r1144539363) in the prior PR. If `.search()` does not find the doc, then it will throw an error here.